### PR TITLE
Improve RArrowDS

### PR DESCRIPTION
**Pull request by ktf**
* Comment out asserts.
* Speedup inner loop by moving some checks one level up in the outer loop.
* Actually support rewinding an RArrowDS

### CODE REVIEW

File: tree/dataframe/src/RArrowDS.cxx [hunk 1]
```diff
@@ -199,6 +199,14 @@ class ArrayPtrVisitor : public ::arrow::ArrayVisitor {
...
 };
 
+#if defined(__GNUC__) || defined(__clang__)
+#define likely(x)       __builtin_expect((x),1)
+#define unlikely(x)     __builtin_expect((x),0)
+#else
+#define likely(x)       x
+#define unlikely(x)     x
+#endif
+
```
Comment (bluehood) [inline tree/dataframe/src/RArrowDS.cxx]: I think ROOT has its own definitions for these @pcanal ?

File: tree/dataframe/src/RArrowDS.cxx [hunk 2]
```diff
@@ -241,46 +249,46 @@ class TValueGetter {
...
+   arrow::Status UncachedSlotLookup(unsigned int slot, ULong64_t entry)
    {
       // If entry is greater than the previous one,
       // we can skip all the chunks before the last one we
       // queried.
       size_t ci = 0;
-      assert(slot < fLastChunkPerSlot.size());
-      if (fLastEntryPerSlot[slot] < entry) {
-         ci = fLastChunkPerSlot.at(slot);
+      // assert(slot < fLastChunkPerSlot.size());
```
Comment (bluehood) [inline tree/dataframe/src/RArrowDS.cxx]: Why leaving the asserts here, but commented? (Why not just leave them?)

File: tree/dataframe/src/RArrowDS.cxx (hunk 1 above)
Comment (ktf) [reply tree/dataframe/src/RArrowDS.cxx]: Fixed.

File: tree/dataframe/src/RArrowDS.cxx (hunk 2 above)
Comment (ktf) [reply tree/dataframe/src/RArrowDS.cxx]: Added back.

**Comment by eguiraud** [comment]
Hi @ktf , I'm finally back on RDF duty. I think this can just go in if you are still interested in having the changes in ROOT. The conflict is trivial to resolve but I don't think I have permission to do a rebase of your branch.

**Comment by eguiraud** [comment]
@ktf ping

**Comment by eguiraud** [comment]
Will merge these changes as part of #3443 .

**Pull request closed without merging**